### PR TITLE
testing mode for writing mails to a file instead of sending them

### PR DESCRIPTION
**Pull request by manukall**
I added an option for mailgun to output mails to a local file instead of actually sending them. I use that for integration testing a single page app, where the tests are run by another process.

### CODE REVIEW

File: lib/client.ex [hunk 1]
```diff
@@ -17,6 +17,14 @@ defmodule Mailgun.Client do
   end
 
   def send_email(conf, email) do
+    if conf[:mode] == :testing do
```
Comment (chrismccord) [inline lib/client.ex]: Let's change this to do_send_email(conf[:mode], ...

then we can just pattern match mulitple clauses of `do_send_email(:test, ..` and skip the if else dance

**Comment by chrismccord** [comment]
Looks great! I have just one small comment

**Comment by manukall** [comment]
I pushed the changes. Thanks for reviewing so quickly.

**Comment by chrismccord** [comment]
Let's make the mode `:test` instead of `:testing` and think this is ready to go

**Comment by manukall** [comment]
do you want the `testing_file_path` to become `test_file_path` then, too?

**Comment by chrismccord** [comment]
yes please

**Comment by manukall** [comment]
ok, changed those.

**Comment by chrismccord** [comment]
<3 <3 <3

**Pull request merged**